source/common/http/conn_manager_impl.cc

```diff
@@ -289,10 +289,12 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
       accumulated_requests_ >= config_.maxRequestsPerConnection()) {
     if (codec_->protocol() < Protocol::Http2) {
       new_stream->state_.saw_connection_close_ = true;
-      drain_state_ = DrainState::Draining;
+      // The response requires a connection close header.
```

Comment:
I don't think this comment makes sense per our previous discussion. From my read of the code it would get a connection close header, but it would also print out this log: 

https://github.com/envoyproxy/envoy/blob/eb18f0b4338179c980680d1f448bcdb3fb043668/source/common/http/conn_manager_impl.cc#L1401

Either way the connection close header would then get added here:

https://github.com/envoyproxy/envoy/blob/eb18f0b4338179c980680d1f448bcdb3fb043668/source/common/http/conn_manager_impl.cc#L1423

So I would just make the comment:

```suggestion
      // Prevent erroneous debug log of closing due to incoming connection close header.
```

Thanks.

/wait